### PR TITLE
[Pipeline] Support pass pipeline input to environment_variables

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
```
@dsl.pipeline(
    name="pass_pipeline_inpute_to_environment_variables",
)
def pipeline(job_in_number: int, environment_variables: str):
    hello_world_component = component_func(component_in_number=job_in_number)
    hello_world_component.environment_variables = environment_variables
```
Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/parameterized_command.py [hunk 1]
```diff
@@ -30,6 +31,11 @@ class ParameterizedCommandSchema(PathAwareSchema):
         ],
         required=True,
     )
-    environment_variables = fields.Dict(keys=fields.Str(), values=fields.Str())
+    environment_variables = UnionField(
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/parameterized_command.py:34]: any other component type need to update?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/parameterized_command.py [hunk 2]
```diff
@@ -30,6 +31,11 @@ class ParameterizedCommandSchema(PathAwareSchema):
...
+        [
+            fields.Dict(keys=fields.Str(), values=fields.Str()),
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/parameterized_command.py:36]: add comment that this is for binding

Comment (lalala123123) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/parameterized_command.py:36]: Add the comment

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/parameterized_command.py (hunk 1 above)
Comment (lalala123123) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/parameterized_command.py:34]: update environment_variables in ParameterizedParallelSchema

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**